File: api/src/main/java/me/iska/jclient/user/Group.java
package me.iska.jclient.user;

import java.util.ArrayList;
import java.util.List;

/**
 * A clientside representation of a group.
 */
public class Group {

    private final List<Permission> permissions = new ArrayList<>();

    private final String name;
    private final int groupID;

    public Group(String name, int groupID, List<Permission> permissions) {
        this.name = name;
        this.groupID = groupID;
        this.permissions.addAll(permissions);
    }

    @Override
    public String toString() {
        return String.format("Group(name=%s, gid=%d)", name, groupID);
    }

    public List<Permission> getPermissions() {
        return new ArrayList<>(permissions);
    }

    public String getName() {
        return name;
    }

    public int getID() {
        return groupID;
    }
}

File: viewer/src/viewer/pyclient/impl/group.py
#!/usr/bin/env python3
from typing import List


class Group:
    """
    A clientside representation of a group.
    """

    @property
    def name(self) -> str:
        return self._name

    @property
    def group_id(self) -> int:
        return self._group_id

    @property
    def permissions(self) -> List:  # -> List[Permission]:
        return self._permissions.copy()

    def __init__(self, name: str, group_id: int, permissions: List) -> None:
        self._name = name
        self._group_id = group_id
        self._permissions = permissions.copy()

    def __repr__(self) -> str:
        return "Group(name=%r, gid=%i)" % (self._name, self._group_id)
